__tests__/patch.js

```diff
@@ -388,6 +581,72 @@ describe("arrays - splice (shrink)", () => {
 	)
 })
 
+describe("sets - add - 1", () => {
+	runPatchTest(
+		new Set([1]),
+		d => {
+			d.add(2)
+		},
+		[{op: "add", path: [1], value: 2}],
+		[{op: "remove", path: [1], value: 2}]
+	)
+})
+
+describe("sets - add, delete, add - 1", () => {
+	runPatchTest(
+		new Set([1]),
+		d => {
+			d.add(2)
+			d.delete(2)
+			d.add(2)
+		},
+		[{op: "add", path: [1], value: 2}],
+		[{op: "remove", path: [1], value: 2}]
+	)
+})
+
+describe("sets - add, delete, add - 2", () => {
+	runPatchTest(
+		new Set([2, 1]),
+		d => {
+			d.add(2)
+			d.delete(2)
+			d.add(2)
+		},
+		[],
+		[]
+	)
+})
+
+describe("sets - mutate - 1", () => {
+	const findById = (set, id) => {
+		for (const item of set) {
+			if (item.id === id) return item
+		}
+	}
+	runPatchTest(
+		new Set([{id: 1, val: "We"}, {id: 2, val: "will"}]),
+		d => {
+			const obj1 = findById(d, 1)
+			const obj2 = findById(d, 2)
+			obj1.val = "rock"
+			obj2.val = "you"
+		},
+		[
+			{op: "remove", path: [0], value: {id: 1, val: "We"}},
```

Comment:
I'd expect patches here that capture that the modification was only the object inside the sets, rather than patches on the set itself, like:
`op: "replace", path: [1, "val"], value: "will"`

Is there a special reason the entire set entry gets reconstructed? (I agree that the concept of indices in a set can be challenged, and maybe your approach is more correct, but sets guarantee insertion order, so I think it should be ok rely on paths in sets?)

Reply:
I think this quite an edge case anyway, so we could also open it as a fresh follow up issue, and see if anybody runs into the difference, before doing anything.

Reply:
Ambiguity of indices in Sets was the primary reason. As you said, you can iterate over them, however, you cannot get an item at index.
Love the idea of leaving this edge case out of the PR.

Reply:
See #450